src/components/taskCreator/ParticipantsSelection.vue

```diff
@@ -0,0 +1,48 @@
+<script setup lang="ts">
+import { ref, computed, watch } from 'vue';
+import { useLtcQuestStore__isTeam } from '@/stores/ltcQuestStore';
+import { teams } from '@/assets/data/teams';
+import { users } from '@/assets/data/users';
+import { useI18n } from "vue-i18n";
+
+const { t } = useI18n();
+
+const ltcQuestStore__isTeam = useLtcQuestStore__isTeam();
+
+const items = computed(() => {
+    if (ltcQuestStore__isTeam.isTeamMode) {
+        return teams.map(team => ({
+            value: team.teamId,
+            label: team.name,
+        }));
+    } else {
+        return users.map(user => ({
+            value: user.userId,
+            label: user.name,
+        }));
+    }
+});
+
+const selectedItems = ref<number[]>([]);
+
+watch(
+    () => ltcQuestStore__isTeam.isTeamMode,
+    (isTeamMode) => {
+        if (isTeamMode) {
+            selectedItems.value = teams.map(team => team.teamId);
+        } else {
+            selectedItems.value = [];
+        }
+    },
+    { immediate: true }
+);
+
+const label = computed(() => {
+    return ltcQuestStore__isTeam.isTeamMode ? t("taskCreator.assignToTeams") : t("taskCreator.assignToParticipants");
+});
+</script>
+
+<template>
+    <v-select :items="items" item-value="value" item-title="label" :label="label" v-model="selectedItems" multiple
+        :rules="[(v: string[]) => v.length > 0 || $t('taskCreator.requirementsValidation')]"></v-select>
```

Comment:
Lepiej przeniesc zawartosc `rules` do `script`a.  
```suggestion
        :rules="[validateSelection]" />
```
i
`const validateSelection = (v: string[]) => {
    const errorMessage = $t('taskCreator.requirementsValidation');
    return v.length > 0 || errorMessage;
};`
Jest wtedy czytelniej, trzymamy się modularności i łatwiej to będzie potem rozwinąć